.github/workflows/test_azure.yml

```diff
@@ -0,0 +1,73 @@
+# This workflow will install Python dependencies and run a basic test to ensure
+# that the Cloud/Azure CLI is still working
+name: Test Azure CLI
+
+on:
+  push:
+    branches: ["main"]
+    paths:
+      - src/mlagility/cli/cloud/**
+  pull_request:
+    branches: ["main"]
+    paths:
+      - src/mlagility/cli/cloud/**
+
+permissions:
+  contents: read
+
+jobs:
+  build_and_test:
+    timeout-minutes: 30
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          miniconda-version: "latest"
+          activate-environment: mla
+          python-version: "3.8"
+      - name: Install dependencies
+        shell: bash -el {0}
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y azure-cli
+          python -m pip install --upgrade pip
+          pip install -e .
+          pip install -r src/mlagility/cli/cloud/requirements.txt
+      - name: Run tests
+        env:
+          SSH_AUTH_SOCK: /tmp/ssh_agent.sock
+          MLAGILITY_AZURE_PUBLIC_KEY: ${{ secrets.AZURE_SSH_PUBLIC_KEY }}
+          AZURE_SUBSCRIPTION_ID: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+          MLAGILITY_AZURE_PASSWORD: ${{ secrets.MLAGILITY_AZURE_PASSWORD }}
+        shell: bash -el {0}
+        run: |
+          az login --username ${{ secrets.AZURE_CLIENT_ID }} --password ${{ secrets.AZURE_CLIENT_PWD }}
+
+          ssh-agent -a $SSH_AUTH_SOCK > /dev/null
+          ssh-add - <<< "${{ secrets.AZURE_SSH_PRIVATE_KEY }}"
+
+          mkdir ~/.ssh
+          touch ~/.ssh/known_hosts
+
+          cd src/mlagility/cli/cloud
+
+          python control.py create setup selftest info stop --name github-test-azure-single --hardware cpu-small
+
+          python control.py create setup selftest info stop --cluster --name github-test-azure-cluster --size 2 --hardware cpu-small
+          python control.py start start run --cluster --name github-test-azure-cluster --size 2 --hardware cpu-small --input-files models/selftest/*.py
+  clean_up:
+    needs: build_and_test
+    if: always()
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install az cli
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y azure-cli
```

Comment:
nit: do we want to specify a version for azure cli?
(Cant be done through package manager, will have to use `wget` followed by `dpkg -i`)

Reply:
IMO we should leave it alone. The commands I'm using in this Action are very basic and I wouldn't expect them to be deprecated.

Reply:
I think the `sudo apt-get update` implies that we'll be getting something fresh (as opposed to something ancient and bad)